src/fides/api/schemas/connection_configuration/connection_secrets_s3.py

```diff
@@ -0,0 +1,54 @@
+from typing import Dict, List, Optional
+
+from pydantic import Field, root_validator
+
+from fides.api.schemas.base_class import NoValidationSchema
+from fides.api.schemas.connection_configuration.connection_secrets import (
+    ConnectionConfigSecretsSchema,
+)
+from fides.api.schemas.storage.storage import AWSAuthMethod
+
+
+class S3Schema(ConnectionConfigSecretsSchema):
+    """Schema to validate the secrets needed to connect to Amazon S3"""
+
+    auth_method: AWSAuthMethod = Field(
```

Comment:
I got an unhandled 500 error when attempting to add an invalid authentication method for S3:

```
2024-06-13 20:23:45.752 | INFO     | fides.api.util.connection_util:validate_secrets:112 - Validating secrets on connection config with key 's3'
2024-06-13 20:23:45.752 | ERROR    | fides.api.util.logger:_log_exception:32 - Object of type AWSAuthMethod is not JSON serializable
```

Reply:
Is there an existing way to prompt to the end user what their options are here?  Seems like an improvement we might ticket

Reply:
oof that error isn't good! let me follow up on that.

in terms of providing more explicit guidance for the end users, i completely agree - we're building out the new "integrations" UI for S3 as we speak, so hopefully that can be accommodated there - we are typing this pretty strongly/explicitly on the BE so it feels like that _should_ be do-able...